frame/offences/benchmarking/src/lib.rs

```diff
@@ -18,44 +18,87 @@
 use pallet_offences::{Trait as OffencesTrait, Module as Offences};
+use pallet_session::historical::{Trait as HistoricalTrait, IdentificationTuple};
+use pallet_session::{Trait as SessionTrait, SessionManager};
 use pallet_staking::{
 	Module as Staking, Trait as StakingTrait, RewardDestination, ValidatorPrefs,
-	Exposure, IndividualExposure, ElectionStatus
+	Exposure, IndividualExposure, ElectionStatus, MAX_NOMINATIONS, Event as StakingEvent
 };
-use pallet_session::Trait as SessionTrait;
-use pallet_session::historical::{Trait as HistoricalTrait, IdentificationTuple};
 
 const SEED: u32 = 0;
 
-const MAX_USERS: u32 = 1000;
 const MAX_REPORTERS: u32 = 100;
 const MAX_OFFENDERS: u32 = 100;
 const MAX_NOMINATORS: u32 = 100;
 const MAX_DEFERRED_OFFENCES: u32 = 100;
 
 pub struct Module<T: Trait>(Offences<T>);
 
-pub trait Trait: SessionTrait + StakingTrait + OffencesTrait + ImOnlineTrait + HistoricalTrait {}
+pub trait Trait:
+	SessionTrait
+	+ StakingTrait
+	+ OffencesTrait
+	+ ImOnlineTrait
+	+ HistoricalTrait
+	+ BalancesTrait
+	+ IdTupleConvert<Self>
+{}
+
+/// A helper trait to make sure we can convert `IdentificationTuple` coming from historical
+/// and the one required by offences.
+pub trait IdTupleConvert<T: HistoricalTrait + OffencesTrait> {
+	/// Convert identification tuple from `historical` trait to the one expected by `offences`.
+	fn convert(id: IdentificationTuple<T>) -> <T as OffencesTrait>::IdentificationTuple;
+}
 
-fn create_offender<T: Trait>(n: u32, nominators: u32) -> Result<T::AccountId, &'static str> {
+impl<T: HistoricalTrait + OffencesTrait> IdTupleConvert<T> for T where
+	<T as OffencesTrait>::IdentificationTuple: From<IdentificationTuple<T>>
+{
+	fn convert(id: IdentificationTuple<T>) -> <T as OffencesTrait>::IdentificationTuple {
+		id.into()
+	}
+}
+
+type LookupSourceOf<T> = <<T as SystemTrait>::Lookup as StaticLookup>::Source;
+type BalanceOf<T> = <<T as StakingTrait>::Currency as Currency<<T as SystemTrait>::AccountId>>::Balance;
+
+struct Offender<T: Trait> {
+	pub controller: T::AccountId,
+	pub stash: T::AccountId,
+	pub nominator_stashes: Vec<T::AccountId>,
+}
+
+fn bond_amount<T: Trait>() -> BalanceOf<T> {
+	T::Currency::minimum_balance().saturating_mul(10_000.into())
+}
+
+fn create_offender<T: Trait>(n: u32, nominators: u32) -> Result<Offender<T>, &'static str> {
 	let stash: T::AccountId = account("stash", n, SEED);
 	let controller: T::AccountId = account("controller", n, SEED);
-	let controller_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(controller.clone());
+	let controller_lookup: LookupSourceOf<T> = T::Lookup::unlookup(controller.clone());
 	let reward_destination = RewardDestination::Staked;
-	let amount = T::Currency::minimum_balance();
-
+	let raw_amount = bond_amount::<T>();
+	// add twice as much balance to prevent the account from being killed.
+	let free_amount = raw_amount.saturating_mul(2.into());
+	T::Currency::make_free_balance_be(&stash, free_amount);
```

Comment:
just a note for whoever might read this in the future and want to work on benchmarks: we use this function quite a lot, and it does not change `T::Currency::issuance`. And we use total issuance in staking's phragmen, and it can cause some very hard to track bugs... so keep that in mind.